src/main/java/spoon/support/compiler/jdt/JDTBasedSpoonCompiler.java

```diff
@@ -390,11 +395,6 @@ protected boolean buildUnitsAndModel(JDTBuilder jdtBuilder, SpoonFolder sourcesF
 		getFactory().getEnvironment().debugMessage(debugMessagePrefix + "build args: " + Arrays.toString(args));
 		batchCompiler.configure(args);
 
-		if (buildOnlyOutdatedFiles && outputDirectory.exists()) {
```

Comment:
I'm not sure about moving this block up: we may need to give the whole bunch of files to JDT in order to avoid compilation error such as missing type, particularly in full-classpath mode.

Reply:
I agree with You! But the problem is that without that move the call of keepOutdatedFiles makes no sense at all because nobody is interested in the result of that call... 
... the work with outdated files is broken in Spoon yet ... it is not only problem of unmodifiable collection ...

Reply:
So, should I write another test or something?

Reply:
> So, should I write another test or something?

what next? I think we need a concept which defines how incremental build has to work. There are many ways how it might be done. @Egor18 I suggest that you define concept which fits exactly to your needs and then in case when the concept makes sense for other clients (not only for You), there is good chance that we implement it once. Or you can then provide a PR which implements this concept - we will gladly support you with that!